monitoring/mock_uss/f3548v21/flight_planning.py

```
@@ -533,17 +540,42 @@ def delete_op_intent(
         op_intent_ref.id,
         op_intent_ref.ovn,
     )
+    return notify_subscribers(
+        result.operational_intent_reference.id, None, result.subscribers, log
+    )
 
+
+def notify_subscribers(
+    op_intent_id: f3548_v21.EntityID,
+    op_intent: Optional[f3548_v21.OperationalIntent],
+    subscribers: List[f3548_v21.SubscriberToNotify],
+    log: Callable[[str], None],
+) -> Dict[f3548_v21.SubscriptionUssBaseURL, Exception]:
+    """
+    Notify subscribers of a changed or deleted operational intent.
+    This function will attempt all notifications, even if some of them fail.
+
+    :return: Notification errors if any, by subscriber.
+    """
+    notif_errors: Dict[f3548_v21.SubscriptionUssBaseURL, Exception] = {}
     base_url = "{}/mock/scd".format(webapp.config[KEY_BASE_URL])
-    for subscriber in result.subscribers:
+    for subscriber in subscribers:
         if subscriber.uss_base_url == base_url:
             # Do not notify ourselves
             continue
         update = f3548_v21.PutOperationalIntentDetailsParameters(
-            operational_intent_id=result.operational_intent_reference.id,
+            operational_intent_id=op_intent_id,
+            operational_intent=op_intent,
             subscriptions=subscriber.subscriptions,
         )
         log(f"Notifying {subscriber.uss_base_url}")
-        scd_client.notify_operational_intent_details_changed(
-            utm_client, subscriber.uss_base_url, update
-        )
+        try:
+            scd_client.notify_operational_intent_details_changed(
+                utm_client, subscriber.uss_base_url, update
+            )
+        except Exception as e:
```

`Exception` is probably too broad as it would even capture, e.g., `KeyboardInterrupt` if the user hit ctrl-c.  Ideally, we would name all (and only) the exceptions that we're confident were due to something we understand and therefore we can recover from and continue normal execution.  In the rare circumstances where we actually want to catch all exceptions, `KeyboardInterrupt` [should be omitted](https://github.com/interuss/monitoring/blob/190def8ac7497ab0ebc45c6f6dcb2b6b4cf61083/monitoring/uss_qualifier/suites/suite.py#L158).

We may want to consider starting to list the Exception types that functions and methods raise "intentionally" (i.e., the candidates for catching here) -- see "Raises" [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).  Not to suggest we do that in this PR, but food for thought for later.

---

So actually I had put this `except Exception` after seeing `Exception` doc: *Common base class for all non-exit exceptions.* Indeed `KeyboardInterrupt` inherits from `BaseException`.

But in any case I agree that it is not good practice to catch all exceptions that way. So I will change that to catch the exceptions that are currently caught in `routes_injection.py`.

> We may want to consider starting to list the Exception types that functions and methods raise "intentionally" (i.e., the candidates for catching here) -- see "Raises" [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). Not to suggest we do that in this PR, but food for thought for later.

That would be nice indeed. Actually one function that would be very useful to document well in that regard is `query_and_describe`. Looking at its implementation it looks like it is supposed to raise only `QueryError`, since it seems to catch `requests.Timeout`, `urllib3.exceptions.ReadTimeoutError` and `requests.RequestException`. But then in the `routes_injection.py` I see that `ConnectionError` and `requests.exceptions.ConnectionError` are caught. So I'm not sure what is supposed to come from where?
And then there is also the `ValueError` that is caught, but I assume that is to cover the parsing errors when trying to access an non-existing field.